File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxCount = 0
        currCount = 0

        for i in nums:
            if i == 1:
                currCount += 1
                maxCount = max(currCount , maxCount)
            else:
                currCount = 0
        
        return maxCount

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int maxOne = 0;
        int currcount = 0;

        for(int i=0; i<n; i++){
            if(nums[i] == 1){
                currcount++;
                maxOne = max(maxOne , currcount);
            }
            else{
                currcount = 0;
            }
        }
        return maxOne;
    }
};